### PR TITLE
Feature/435 page break nav

### PR DESCRIPTION
**Pull request by sbrody**
## Description of changes
...
Implements in-page navigation for long read articles created using the page break method. The current page is found using a search of page content for page breaks and a use of a query variable.

The functionality can be tested by passing new \LongReadPlugin\PageBreakInPageNavigation() as the second argument of the Navigation instance in di.php,

## Checklist
- [x] Changelog updated
- [ ] If new release: major version tag to be bumped after release (see [docs](../README.md#changelog-and-versioning))

### CODE REVIEW

File: spec/page_break_in_page_navigation.spec.php [hunk 1]
```diff
@@ -0,0 +1,146 @@
+<?php
+
+describe(LongReadPlugin\PageBreakInPageNavigation::class, function () {
```
Comment (RobjS) [inline spec/page_break_in_page_navigation.spec.php:3]: Can we add some tests here to show how the in page navigation works in terms of only selecting items that are in the current page?

File: src/PageBreakInPageNavigation.php [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+	private $inPageNavItems = [];
+
+	private function getCurrentPageMarkup(): string
+	{
+		global $post;
+		$currentPage = max(1, (int) get_query_var('page', 1));
+		$fullContent = $post->post_content ?? '';
+		$pages = explode('<!--nextpage-->', $fullContent);
+		return $pages[$currentPage - 1] ?? '';
+	}
```
Comment (RobjS) [inline src/PageBreakInPageNavigation.php:29]: This method aside, this class is essentially the same as the `HierarchicalInPageNavigation` class. I think that's fine for now, but there's probably room for a refactor in future to remove some of that duplication.

**Review by RobjS — CHANGES_REQUESTED** [review]
This is working for me locally. I think it'd be good to have some tests showing how only the in page items from the current page are selected. No need to do anything about removing the duplication between the in page nav classes for now, just worth noting for future reference.

File: spec/page_break_in_page_navigation.spec.php (hunk 1 above)
Comment (sbrody) [reply spec/page_break_in_page_navigation.spec.php:3]: Yes - have added those in

File: src/PageBreakInPageNavigation.php (hunk 2 above)
Comment (sbrody) [reply src/PageBreakInPageNavigation.php:29]: Refactor sounds like a good idea as I'm coming up against more repeating code.
I've also added some new fallback functionality as I noticed that in-page headers on pages beyond the first was sometimes not showing

File: spec/page_break_in_page_navigation.spec.php [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+
+			$result = $this->inPageNavigation->getItems();
+
+			expect(count($result))->toEqual(1);
+			expect($result[0])->toBeAnInstanceOf(\LongReadPlugin\InPageNavigationItem::class);
+			expect($result[0]->title)->toEqual('ACF grouped heading');
+			expect($result[0]->id)->toEqual('acf-grouped');
+		});
+
+		it('parses only the current page markup from post content', function () {
```
Comment (RobjS) [inline spec/page_break_in_page_navigation.spec.php:146]: I think this test could arguable be removed, as it's really testing the internal implementation (i.e. how `parse_blocks()` is called) rather than the external behaviour of `getItems()`. And the behaviour of `getItems()`  across multiple pages is described in a later test.

**Review by RobjS — APPROVED** [review]
This is behaving as described for me locally. Well done on spotting that edge case that requires a fallback. 

One minor comment on the tests, but nothing that needs to block merging.

File: spec/page_break_in_page_navigation.spec.php (hunk 3 above)
Comment (sbrody) [reply spec/page_break_in_page_navigation.spec.php:146]: Ok cheers - will remove in a later PR
